File: front_igm/CustomerView/models/Subscription.js
import { Plan } from "./Plan.js";
import { Shop } from "./Shop.js";

export class Subscription {
    static STATE = ["waiting", "paid", "expired"];

    constructor(id, shop, plan, cantProducts, state, until_date) {
        this.id = id; // string
        this.shop = shop; // Shop
        this.plan = plan; // Plan
        this.cantProducts = cantProducts; // number
        this.state = Subscription.STATE[state]; // number index
        this.until_date = until_date; // string | Date
    }

    static fromJson(json) {
        if (!json) return null;

        return new Subscription(
            String(json.id),
            Shop.fromJson(json.shop),
            Plan.fromJson(json.plan),
            Number(json.cantProducts),
            Number(json.state),
            json.until_date
        );
    }

    toJson() {
        return {
            id: this.id,
            shop: this.shop ? this.shop.toJson() : null,
            plan: this.plan ? this.plan.toJson() : null,
            cantProducts: this.cantProducts,
            state: Subscription.STATE.indexOf(this.state),
            until_date: this.until_date
        };
    }
}

File: back_igm/core/models/subscription.py
from plan import Plan
from shop import Shop
from typing import Dict, Any

class Subscription:
    STATE = ["waiting", "paid", "expired"]

    def __init__(self, id: str, shop: Shop, plan: Plan,
                 cantProducts: int, state: int, until_date):
        self.id = id
        self.shop = shop
        self.plan = plan
        self.cantProducts = cantProducts
        self.state = Subscription.STATE[state]
        self.until_date = until_date

    @staticmethod
    def fromJson(data: Dict[str, Any]) -> "Subscription":
        return Subscription(
            id=str(data["id"]),
            shop=Shop.fromJson(data["shop"]),
            plan=Plan.fromJson(data["plan"]),
            cantProducts=int(data["cantProducts"]),
            state=int(data["state"]),
            until_date=data["until_date"]
        )

    def toJson(self) -> Dict[str, Any]:
        return {
            "id": self.id,
            "shop": self.shop.toJson() if self.shop else None,
            "plan": self.plan.toJson() if self.plan else None,
            "cantProducts": self.cantProducts,
            "state": Subscription.STATE.index(self.state),
            "until_date": self.until_date
        }